File: InterviewPrep/day1/removeElement.py
# https://leetcode.com/problems/remove-element
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        j=0
        for i in nums:
            if i != val:
                nums[j] = i
                j += 1
        return j

File: InterviewPrep/day1/removeElement.cpp
// https://leetcode.com/problems/remove-element

class Solution {
	public:
		int removeElement(vector<int>& nums, int val) {
			int arrIdx = 0;
			int size = nums.size();

			for (int i = 0; i < size; ++i) {
				if (nums[i] != val) {                
					if (arrIdx != i) {
						nums[arrIdx] ^= nums[i];
						nums[i] ^= nums[arrIdx];
						nums[arrIdx] ^= nums[i];
					}
					arrIdx++;
				}        
			}
			return arrIdx;
		}
};
